File: backend/internal/models/treatment.go
package models

import "time"

type Treatment struct {
	ID            string             `json:"id"`
	Slug          string             `json:"slug"`
	NameDE        string             `json:"name_de"`
	NameEN        string             `json:"name_en"`
	DescriptionDE string             `json:"description_de,omitempty"`
	DescriptionEN string             `json:"description_en,omitempty"`
	Material      string             `json:"material,omitempty"`
	DurationDays  int                `json:"duration_days"`
	LifespanYears int                `json:"lifespan_years"`
	Advantages    []TreatmentAdvantage `json:"advantages"`
	Pricing       TreatmentPricing   `json:"pricing"`
	SortOrder     int                `json:"-"`
	Active        bool               `json:"-"`
	CreatedAt     time.Time          `json:"-"`
	UpdatedAt     time.Time          `json:"-"`
}

type TreatmentAdvantage struct {
	IconDE  string `json:"icon_de"`
	TitleDE string `json:"title_de"`
	TitleEN string `json:"title_en"`
	DescDE  string `json:"desc_de"`
	DescEN  string `json:"desc_en"`
}

type TreatmentPricing struct {
	AlbaniaPerTooth    int            `json:"albania_per_tooth_eur"`
	GermanyAvgPerTooth int            `json:"germany_avg_per_tooth_eur"`
	PackagePrices      []PackagePrice `json:"package_prices"`
}

type PackagePrice struct {
	NameDE    string `json:"name_de"`
	NameEN    string `json:"name_en"`
	Teeth     int    `json:"teeth_count"`
	Price     int    `json:"price_eur"`
}

File: frontend/src/types/treatment.ts
export interface TreatmentAdvantage {
  icon_de: string;
  title_de: string;
  title_en: string;
  desc_de: string;
  desc_en: string;
}

export interface PackagePrice {
  name_de: string;
  name_en: string;
  teeth_count: number;
  price_eur: number;
}

export interface TreatmentPricing {
  albania_per_tooth_eur: number;
  germany_avg_per_tooth_eur: number;
  package_prices: PackagePrice[];
}

export interface Treatment {
  id: string;
  slug: string;
  name_de: string;
  name_en: string;
  description_de?: string;
  description_en?: string;
  material?: string;
  duration_days: number;
  lifespan_years: number;
  advantages: TreatmentAdvantage[];
  pricing: TreatmentPricing;
}
